packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -261,29 +261,22 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             blob = response.content;
         }
 
+        let documentAttributes: api.IDocumentAttributes = JSON.parse(blob.content);
+        if (blob.encoding === "base64") {
+            documentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
+        }
+
         if (this.attributesBlobHandles.has(blobid)) {
             // ODSP document ids are random guids (different per session)
             // fix the branch name in attributes
             // this prevents issues when generating summaries
-            if (blob.encoding === "base64") {
-                const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
-                documentAttributes.branch = this.documentId;
-
-                if (outputFormat === "base64") {
-                    blob.content = fromUtf8ToBase64(JSON.stringify(documentAttributes));
-                } else {
-                    blob.content = JSON.stringify(documentAttributes);
-                }
-            } else {
-                const documentAttributes: api.IDocumentAttributes = JSON.parse(blob.content);
-                documentAttributes.branch = this.documentId;
+            documentAttributes.branch = this.documentId;
+        }
 
-                if (outputFormat === "base64") {
-                    blob.content = fromUtf8ToBase64(JSON.stringify(documentAttributes));
-                } else {
-                    blob.content = JSON.stringify(documentAttributes);
-                }
-            }
+        if (outputFormat === "base64") {
```

Comment:
Theoretically, same blob can be read multiple times through different APIs. Given that we modify blob.content, we also need to update blob.encoding to reflect format of the data we are putting in.
I think better solution would be just return data in right format without changing blob.content